core/domain/stats_services_test.py

```diff
@@ -1151,6 +1159,223 @@ def test_cst_exp_issue_is_updated_when_state_is_renamed(self):
+
+        exp_issue = exp_issues.unresolved_issues[0]
+
+        model = stats_models.PlaythroughModel.get_by_id(
+            exp_issue.playthrough_ids[0])
+
+        playthrough = stats_services.get_playthrough_from_model(model)
+
+        self.assertLess(
+            len(exp_issues.unresolved_issues[0].playthrough_ids),
+            feconf.MAX_PLAYTHROUGHS_FOR_ISSUE)
+
+        stats_services.assign_playthrough_to_corresponding_issue(
+            playthrough, exp_issues, stats_models.CURRENT_ISSUE_SCHEMA_VERSION)
+
+    def test_assign_playthrough_to_corresponding_issue_is_false(self):
+        stats_services.save_exp_issues_model(
+            stats_domain.ExplorationIssues(self.exp.id, self.exp.version, [
+                self._create_cst_exp_issue(
+                    [self._create_cst_playthrough(['A', 'B', 'A'])],
+                    ['A', 'B', 'A'])
+            ]))
+
+        exp_issues = (
+            stats_services.get_exp_issues(self.exp.id, self.exp.version))
+
+        exp_issue = exp_issues.unresolved_issues[0]
+
+        model = stats_models.PlaythroughModel.get_by_id(
+            exp_issue.playthrough_ids[0])
+
+        playthrough = stats_services.get_playthrough_from_model(model)
+
+        exp_issues.unresolved_issues[0].playthrough_ids = [
+            'playthrough_id1', 'playthrough_id2', 'playthrough_id3',
+            'playthrough_id4', 'playthrough_id5', 'playthrough_id6']
+
+        self.assertTrue(
+            len(exp_issues.unresolved_issues[0].playthrough_ids) >=
+            feconf.MAX_PLAYTHROUGHS_FOR_ISSUE)
+
+        stats_services.assign_playthrough_to_corresponding_issue(
+            playthrough, exp_issues, stats_models.CURRENT_ISSUE_SCHEMA_VERSION)
+
+    def test_get_corresponding_exp_issue(self):
+        stats_services.save_exp_issues_model(
+            stats_domain.ExplorationIssues(self.exp.id, self.exp.version, [
+                self._create_cst_exp_issue(
+                    [self._create_cst_playthrough(['A', 'B', 'A'])],
+                    ['A', 'B', 'A'])
+            ]))
+
+        exp_services.update_exploration(self.owner_id, self.exp.id, [
+            exp_domain.ExplorationChange({
+            'cmd': 'rename_state',
+                'old_state_name': 'A',
+                'new_state_name': 'Z',
+            })
+        ], 'change')
+
+        exp_issues = (
+            stats_services.get_exp_issues(self.exp.id, self.exp.version + 1))
+
+        exp_issue = exp_issues.unresolved_issues[0]
+
+        model = stats_models.PlaythroughModel.get_by_id(
+        exp_issue.playthrough_ids[0])
+
+        playthrough = stats_services.get_playthrough_from_model(model)
+
+        exp_issues = stats_domain.ExplorationIssues(
+            self.exp.id, stats_models.CURRENT_ISSUE_SCHEMA_VERSION, [])
+
+        stats_services.assign_playthrough_to_corresponding_issue(
+        playthrough, exp_issues, stats_models.CURRENT_ISSUE_SCHEMA_VERSION)
```

Comment:
Why do we have this in a test that (at least according to the name) is just testing whether we can retrieve a corresponding exploration issue?

Reply:
assign_playthrough_to_corresponding_issue calls _get_corresponding_exp_issue, but since this is a protected member, we had to call it indirectly.

Reply:
In that case, we shouldn't be framing the test as testing `_get_corresponding_exp_issue` since that's a private method. Instead, we should be testing the public API